src/model/ChillerElectricASHRAE205.cpp

```diff
@@ -0,0 +1,1179 @@
+    unsigned ChillerElectricASHRAE205_Impl::tertiaryInletPort() const {
+      return OS_Chiller_Electric_ASHRAE205Fields::HeatRecoveryInletNodeName;
+    }
+
+    unsigned ChillerElectricASHRAE205_Impl::tertiaryOutletPort() const {
+      return OS_Chiller_Electric_ASHRAE205Fields::HeatRecoveryOutletNodeName;
+    }
+
+    boost::optional<Node> ChillerElectricASHRAE205_Impl::heatRecoveryInletNode() const {
+      if (auto mo_ = tertiaryInletModelObject()) {
+        return mo_->optionalCast<Node>();
+      }
+      return boost::none;
+    }
+
+    boost::optional<Node> ChillerElectricASHRAE205_Impl::heatRecoveryOutletNode() const {
+      if (auto mo_ = tertiaryOutletModelObject()) {
+        return mo_->optionalCast<Node>();
+      }
+      return boost::none;
+    }
+
+    bool ChillerElectricASHRAE205_Impl::addToNode(Node& node) {
+      boost::optional<PlantLoop> t_plantLoop = node.plantLoop();
+
+      // If trying to add to a node that is on the demand side of a plant loop
+      if (t_plantLoop) {
+        if (t_plantLoop->demandComponent(node.handle())) {
+          // If there is already a another loop where it's on the demand side (condenser Plant Loop)
+          boost::optional<PlantLoop> cndLoop_ = this->condenserWaterLoop();
+          if (cndLoop_) {
+            // And it's not the same as the node's loop
+            if (t_plantLoop.get() != cndLoop_.get()) {
+              // And if there is no Heat Recovery (tertiary)
+              if (!this->heatRecoveryLoop().is_initialized()) {
+                // Then try to add it to the tertiary one
+                LOG(Warn, "Calling addToTertiaryNode to connect it to the tertiary (=Heat Recovery) loop for " << briefDescription());
+                return this->addToTertiaryNode(node);
+              }
+            }
+          }
+        }
+      }
+
+      // All other cases, call the base class implementation
+      return WaterToWaterComponent_Impl::addToNode(node);
+    }
+
+    bool ChillerElectricASHRAE205_Impl::addToTertiaryNode(Node& node) {
+
+      if constexpr (!ChillerElectricASHRAE205::isHeatRecoverySupportedByEnergyplus) {
+        LOG(Warn, "For " << briefDescription() << ", Heat Recovery isn't implemented by EnergyPlus yet, so tertiary connection is disabled.");
+        return false;
+      }
+
+      auto t_plantLoop = node.plantLoop();
+
+      // Only accept adding to a node that is on a demand side of a plant loop
+      // Since tertiary here = heat recovery loop (heating)
+      if (t_plantLoop) {
+        if (t_plantLoop->demandComponent(node.handle())) {
+          // Call base class method which accepts both supply and demand
+          return WaterToWaterComponent_Impl::addToTertiaryNode(node);
+        } else {
+          LOG(Info,
+              "Tertiary Loop (Heat Recovery Loop) connections can only be placed on the Demand side (of a Heating Loop), for " << briefDescription());
+        }
+      }
+      return false;
+    }
```

Comment:
Disabled for now, but I have done that last to make sure it works correctly anyways.